roomba_980/__init__.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# vim: set encoding=utf-8 tabstop=4 softtabstop=4 shiftwidth=4 expandtab
+#########################################################################
+#  Copyright 2011 KNX-User-Forum e.V.           http://knx-user-forum.de/
+#########################################################################
+#  This file is part of SmartHome.py.    http://mknx.github.io/smarthome/
+#
+#  SmartHome.py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  SmartHome.py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with SmartHome.py. If not, see <http://www.gnu.org/licenses/>.
+#########################################################################
+
+import logging
+from plugins.roomba_980.roomba import Roomba
+from lib.model.smartplugin import SmartPlugin
+from lib.item import Items
+
+class ROOMBA_980(SmartPlugin):
+
+    ALLOW_MULTIINSTANCE = False
+    PLUGIN_VERSION = "1.0.0"
+
+    logger = logging.getLogger(__name__)
+
+    myroomba = None
+
+    def __init__(self, sh, adress=None, blid=None, roombaPassword=None, cycle=900):
+        self._sh = sh
+        self._address = adress
+        self._blid = blid
+        self._roombaPassword = roombaPassword
+        self._cycle = cycle
+
+        self._status_batterie = None
+        self._status_items = {}
+
+        self.myroomba = Roomba(self._address, self._blid, self._roombaPassword)
+        self.myroomba.connect()
+
+    def parse_logic(self, logic):
+        pass
+
+    def parse_item(self, item):
+        if self.has_iattr(item.conf, 'roomba_980'):
+            item_type = self.get_iattr_value(item.conf, 'roomba_980')
+            if item_type == "start" or item_type == "stop" or item_type == "dock":
+                return self.update_item
+
+            self._status_items[item_type] = item
+
+            self.logger.debug('{} item gefunden {}'.format(item_type, item))
+
+    def run(self):
+        self.scheduler_add(__name__, self.get_status, prio=5, cycle=self._cycle, offset=2)
+        self.alive = True
+
+    def stop(self):
+        self.scheduler.remove(__name__)
+        self.myroomba.disconnect()
+        self.alive = False
+
+    def __call__(self):
+        pass
+
+    def update_item(self, item, caller=None, source=None, dest=None):
+        if caller != __name__ and self.has_iattr(item.conf, 'roomba_980'):
```

Comment:
The `has_iattr()` check is not required here because the `update_item()` method is only called for items which are handled in the `parse_item()` method. This method already checks for the item attribute.

Or is there any other situation where this method get called without having this item configured?